Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -67,43 +79,76 @@ public final class CoreLocationMiddleware: Middleware {
         delegate.output = output
         manager.delegate = delegate
     }
+    
     public func handle(action: LocationAction, from dispatcher: ActionSource, afterReducer: inout AfterReducer) {
         switch action {
-        case .startMonitoring, .authorized:
-            startMonitoring()
-        case .stopMonitoring:
-            stopMonitoring()
+        case .startMonitoring: startMonitoring()
+        case .stopMonitoring: stopMonitoring()
+        case .requestAuthorizationStatus:
+            if #available(iOS 14.0, *) {
+                delegate.output?.dispatch(getAuthzStatus(status: manager.authorizationStatus))
+            } else {
+                return
+            }
+        case .requestAuthorizationType:
+            switch getState?().authzType {
+            case .always:
+                manager.requestAlwaysAuthorization()
+            case .whenInUse:
+                manager.requestWhenInUseAuthorization()
+            case .none:
+                break
+            }
+        case .requestPosition: manager.requestLocation()
+        case .requestDeviceCapabilities: delegate.output?.dispatch(getDeviceCapabilities())
         default: return
         }
     }
+    
     func startMonitoring() {
-        switch getState?() {
-        case .authorized?:
-            manager.startUpdatingLocation()
-        default:
-            // requestAlwaysAuthorization or requestWhenInUseAuthorization could be decided on
-            // the middleware init or as payload for startMonitoring
-            manager.requestAlwaysAuthorization()
+        let stateType = getState?().authzType
+        let stateStatus = getState?().authzStatus
+        
+        switch stateType {
+        case .always:
+            if CLAuthorizationStatus.authorizedAlways != stateStatus {
+                manager.requestAlwaysAuthorization()
```

Comment:
Should we return in this case, to avoid manager.startUpdatingLocation() when we don't have the permission we want?

Reply:
Good question ! Part of me thinks that "it doesn't matter™️". The reason being that the app wants to start the location service (that's what it controls), but has no way to prevent the user from changing the authorization status after the service has started. So would the app stop the service then ?
The app can provide a warning since we're emitting any authorization status change or decide to stop the location service altogether.
Or we could punt on the stateType check and separate those 2 concerns : essentially when you start the service, that's all the Middleware would do, that is call `manager.startUpdatingLocation()` regardless of the status. Then if the app wants to change the authorization status, it can do that off-band and decide whether or not to start the service based on the result of that operation.
So bottom line : 🤷‍♂️ ?
Since you brought that up, I am actually thinking the latter option might work best. But maybe you might be able to see something wrong with that approach...

Reply:
Created a new issue (#6) for this so we can discuss and address it separately.